apps/emqx/src/emqx_channel.erl

```diff
@@ -2322,7 +2324,7 @@ ensure_keepalive_timer(disabled, Channel) ->
 ensure_keepalive_timer(Interval, Channel = #channel{clientinfo = #{zone := Zone}}) ->
     Multiplier = get_mqtt_conf(Zone, keepalive_multiplier),
     RecvCnt = emqx_pd:get_counter(recv_pkt),
-    Keepalive = emqx_keepalive:init(RecvCnt, round(timer:seconds(Interval) * Multiplier)),
+    Keepalive = emqx_keepalive:init(RecvCnt, round(timer:seconds(Interval) * Multiplier / 2)),
```

Comment:
Move this calculation to emqx_keepalive.erl.

Also:
I believe the reason for MQTT specification to define `1.5` as multiplier is to give 0.5 of the interval shift for the checks.
To better align the timer expire even with the intervals, we should check more often, a good default is `Interval div 2`, and allow client to be idle for 3 checks. 

If we hard-code the number of idle checks to 2, the error (误差）becomes bigger when the interval is larger.

In order to allow users to fine-tune the parameters to lower the error (误差）, we should make it possible for users to adjust both the interval, and the number of idle checks.

```
TimerInterval = timer:seconds(Interval) div 2,
MaxIdleInterval = floor(2 * Multiplier + 0.000001),
```

As an example, for default multiplier `1.5` and interval=`12s`, the check is done every `6s` and disconnected after `18s` but before `24s`.

Another example: if one configures `multiplier=9` and `interval=2s`, then the check is done every `1s`, the client is STILL disconnected after `18s` but before `19s`.

I remember discussing this with someone quite a long time ago, but in the end it was implemented wrong for some reason I cannot recall.

Reply:
There's a previous discussion here ? https://github.com/emqx/emqx/issues/11685#issuecomment-1764108957

But, IMO: Make multiplier high and interval too small, although it can be used this way, it is obviously not good(This is not really what the user wants). interval = 1s means that ping requests are sent every second. This is not a recommended design for either the client or the server.

It's a good idea to make this `2` a configurable item.